app/lib/test_applications.rb

```diff
@@ -85,54 +95,112 @@ def self.create_application(states:, courses_to_apply_to: nil)
       MakeAnOffer.new(actor: actor, application_choice: choice, offer_conditions: ['Complete DBS', 'Fitness to teach check']).save
+      choice.update_columns(offered_at: @time)
+      fast_forward(1..3)
       AcceptOffer.new(application_choice: choice).save!
-    elsif state == :accepted_no_conditions
+      choice.update_columns(accepted_at: @time)
+    when :accepted_no_conditions
+      fast_forward(1..3)
       MakeAnOffer.new(actor: actor, application_choice: choice, offer_conditions: []).save
+      choice.update_columns(offered_at: @time)
+      fast_forward(1..3)
       AcceptOffer.new(application_choice: choice).save!
-    elsif state == :conditions_not_met
+      choice.update_columns(accepted_at: @time)
+    when :conditions_not_met
+      fast_forward(1..3)
       MakeAnOffer.new(actor: actor, application_choice: choice, offer_conditions: ['Complete DBS', 'Fitness to teach check', 'Complete course']).save
+      choice.update_columns(offered_at: @time)
+      fast_forward(1..3)
       AcceptOffer.new(application_choice: choice).save!
+      choice.update_columns(accepted_at: @time)
+      fast_forward(1..3)
       ConditionsNotMet.new(application_choice: choice).save
-    elsif state == :recruited
+      choice.update_columns(conditions_not_met_at: @time)
+    when :recruited
+      fast_forward(1..3)
       MakeAnOffer.new(actor: actor, application_choice: choice, offer_conditions: ['Complete DBS', 'Fitness to teach check']).save
+      choice.update_columns(offered_at: @time)
+      fast_forward(1..3)
       AcceptOffer.new(application_choice: choice).save!
+      choice.update_columns(accepted_at: @time)
+      fast_forward(1..3)
       ConfirmOfferConditions.new(application_choice: choice).save
-    elsif state == :enrolled
+      choice.update_columns(recruited_at: @time)
+    when :enrolled
+      fast_forward(1..3)
       MakeAnOffer.new(actor: actor, application_choice: choice, offer_conditions: ['Complete DBS']).save
+      choice.update_columns(offered_at: @time)
+      fast_forward(1..3)
       AcceptOffer.new(application_choice: choice).save!
+      choice.update_columns(accepted_at: @time)
+      fast_forward(1..3)
       ConfirmOfferConditions.new(application_choice: choice).save
+      choice.update_columns(recruited_at: @time)
+      fast_forward(1..3)
       ConfirmEnrolment.new(application_choice: choice).save
-    elsif state == :withdrawn
+      choice.update_columns(enrolled_at: @time)
+    when :withdrawn
+      fast_forward(1..3)
       WithdrawApplication.new(application_choice: choice).save!
+      choice.update_columns(withdrawn_at: @time)
     end
   end
 
-  def self.actor
+  def actor
     SupportUser.first_or_initialize
   end
 
-  def self.without_slack_message_sending
+  def without_slack_message_sending
     RequestStore.store[:disable_slack_messages] = true
     yield
     RequestStore.store[:disable_slack_messages] = false
   end
+
+  def travel_to(time)
+    @time = time
```

Comment:
Is it worth providing an `attr_reader` for this instance var as it's referenced so often?